### PR TITLE
[rosdep/base] Added rosdep keys for individual pcl packages

### LINKED ISSUE #26438: {Proposal} Adding fine-grained PCL dependencies
Dear Developers,
My name is Yoshua Nava. I'm a Perception Software Engineer at ANYbotics AG (Zurich, Switzerland)

I'm writing this issue in relation to https://github.com/ros-perception/perception_pcl/issues/307. As mentioned in the perception_pcl issue, we would like to have more fine-grained PCL dependencies in the perception_pcl package, by adding the corresponding Debians for pcl modules to rosdistro.

Some of the packages that could be added (although I need to confirm this), are:

```
libpcl-common
libpcl-conversions-dev
libpcl-features
libpcl-filters
libpcl-io
libpcl-kdtree
libpcl-keypoints
libpcl-ml
libpcl-msgs-dev
libpcl-octree
libpcl-outofcore
```

I would like to propose adding the corresponding rosdep keys to the base.yaml file. Would the maintainers approve of this initiative? If affirmative I can open the corresponding pull request.

Thank you in advance.

Best regards,
Yoshua Nava

### PR DESCRIPTION
**Pull request by YoshuaNava**
This PR aims to resolve #26438.

For now I have just added Debian and Ubuntu dependencies. An open question: For completeness, should I also add the rest of the existing pcl PPA packages? (i.e. apps, keypoints, ml, octree, outofcore, people, recognition, registration, sample-consensus, search, stereo, surface, tracking and visualization).

PS: As mentioned in the issue, I couldn't find information about pcl modules for OS != {ubuntu, debian}. Thus, we can't depend on individual packages in perception_pcl until we can link against the correct modules.

Best,
Yoshua Nava

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -3379,6 +3379,146 @@ libpcl-all-dev:
...
     zesty: [libpcl-dev]
+libpcl-common:
+  debian: [libpcl-common1.8]
+  ubuntu:
+    artful: [libpcl-common1.8]
+    bionic: [libpcl-common1.8]
+    cosmic: [libpcl-common1.8]
+    disco: [libpcl-common1.8]
+    eoan: [libpcl-common1.8]
+    focal: [libpcl-common1.8]
```
Comment (hidmic) [inline rosdep/base.yaml:3390]: @YoshuaNava hmm, a quick apt cache search shows `libpcl-common1.8` is missing from `universe` and `multiverse` in Focal.

File: rosdep/base.yaml [hunk 2]
```diff
@@ -3379,6 +3379,146 @@ libpcl-all-dev:
...
+    precise: [libpcl-1.7-common1.8]
+    quantal: [libpcl-1.7-common1.8]
+    raring: [libpcl-1.7-common1.8]
+    saucy: [libpcl-1.7-common1.8]
+    trusty: [libpcl-1.7-common1.8]
```
Comment (hidmic) [inline rosdep/base.yaml:3395]: @YoshuaNava hmm, I can't seem to even find these fine grained PCL releases on `trusty`.

**Review by hidmic — COMMENTED** [review]
> For completeness, should I also add the rest of the existing pcl PPA packages?

The more rules the better :smiley: 

Having said that, I'm a bit wary about what you mean by _pcl PPA_. Which PPA are you talking about? [CONTRIBUTING](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#expected-rosdep-sources) guidelines are very explicit about which package sources are allowed.

**Review by tfoote — CHANGES_REQUESTED** [review]
Please be clearer about where and how these new packages are coming from. And in particular, please link to the packages on packages.debian.org and packages.ubuntu.com for us to be able to review in your PR description. 

Since you're trying to optimize dependencies for other platforms you should provide the less optimized solution or depending on the lightest weight package which provides the equivalent functionality. If Fedora as libraries A B and C in one packages you can have a rule for A, B and C separately all depend on the ABC package on that platform.

**Comment by mjcarroll** [comment]
@YoshuaNava Friendly ping.

**Comment by ivanpauno** [comment]
@YoshuaNava friendly ping

**Comment by clalancette** [comment]
There's been no response to the original comments in over a month, so I'm going to close this out.

If you'd like to continue with this PR, please respond to the comment and feel free to reopen the PR.  Thanks for the contribution.

**Pull request closed without merging**